python/iot-rest-api-sample.py

```diff
@@ -32,64 +33,79 @@
 Data is first validated using /validate-beacons end point and then sent to /beacons end point to be stored
 '''
 
-# set iot appKey and collectorUrl
-iot = dict()
-iot['appKey'] = 'AB-AAB-AAC-AAD'
-iot['collectorUrl'] = 'https://iot-col.eum-appdynamics.com'
+# Read Command Line Options
+parser = argparse.ArgumentParser()
 
-iot['validateBeaconUrl'] = iot['collectorUrl'] + '/eumcollector/iot/v1/application/' + iot['appKey'] + '/validate-beacons'
-iot['sendBeaconUrl'] = iot['collectorUrl'] + '/eumcollector/iot/v1/application/' + iot['appKey'] + '/beacons'
+# Mandatory argument - appkey
+parser.add_argument("appkey", help="set application key")
 
-# beacon with custom event
-beacon_custom_event = [{
-  'deviceInfo': {
-    'deviceId': '1111',
-    'deviceName': 'AudiS3',
-    'deviceType': 'SmartCar'
-  },
-  'versionInfo': {
-    'hardwareVersion': '1.0',
-    'firmwareVersion': '1.0',
-    'softwareVersion': '1.0',
-    'operatingSystemVersion': '1.0'
-  },
-  'customEvents': [{
-    'eventType': 'Custom Event',
-    'eventSummary': 'Diagnostic Data Captured in Smart Car',
-    'timestamp': (int(time.time())*1000),
-    'datetimeProperties': {
-      'Last Engine Start Time': 1512483673000
-    },
-    'stringProperties': {
-      'VinNumber': 'VN123456',
-    },
-    'doubleProperties': {
-      'Temperature': 101.3
-    },
-    'longProperties': {
-      'MPG Reading': 23,
-      'Annual Mileage': 12000
-    },
-    'booleanProperties': {
-      'Engine Lights ON': 'false'
-    }
-  }]
-}]
+#optional arguments
+parser.add_argument("-c", "--collectorurl", default = "https://iot-col.eum-appdynamics.com",
+                    help="set IoT Collector URL to which the beacons should be sent to")
+parser.add_argument("-u", "--requesturl", help="set sample URL to trigger network request and capture network event")
+parser.add_argument("-x", "--requesttype", default = "GET", help="set request type for the URL. Default is set to GET")
+parser.add_argument("-d", "--requestdata", help="set data to be sent with HTTP Request for URL")
+parser.add_argument("-v", "--verbose", action="store_true", help="enable debug info")
```

Comment:
TODO: there are some great tools that will take requests in python and convert them to curl, consider adding them, example: https://github.com/ofw/curlify

Reply:
You really should add a compress on/off option

Reply:
Speaking of compression, how will verbose print the binary wad?